model/dataProvider/providers/TestSessionDataProviderService.php

```diff
@@ -0,0 +1,59 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2019  (original work) Open Assessment Technologies SA;
+ */
+
+namespace oat\taoSyncClient\model\dataProvider\providers;
+
+use oat\oatbox\service\ConfigurableService;
+use oat\taoSyncClient\model\dataProvider\SyncPackageDataProviderInterface;
+use oat\taoSyncClient\model\syncQueue\SyncQueueInterface;
+
+class TestSessionDataProviderService extends ConfigurableService implements SyncPackageDataProviderInterface
+{
+
+    public function getData($deliveryExecutionIds = [])
+    {
+        return $this->getValidDeliveryExecutions($deliveryExecutionIds);
+    }
+
+    /**
+     * Preconditions to have possibility to sync test sessions
+     * @param array $deliveryExecutionIds
+     * @return array
+     */
+    private function getValidDeliveryExecutions($deliveryExecutionIds = [])
+    {
+        $deliveryExecutionIds = array_unique($deliveryExecutionIds);
+        foreach ($deliveryExecutionIds as $key => $deliveryExecutionId) {
+            // all delivery logs need to be synchronized before test session sending
+            if (!$this->getSyncQueueService()->isDeliveryLogSynchronized($deliveryExecutionId)) {
```

Comment:
With this approach it looks like we will need to call `SyncPackageService::create()` two times to include delivery_logs into first package and test_sessions into second one, which is not a valid use case. 
We need to include all data which should be synchronized into package.

Reply:
yes, we can run GeneratePackage script with -e parameter and will be generated 2 packages, then syncServer can load both these packages and import everything, but this is safest than put everything into the same package